### PR TITLE
Loading bar for fetch-every-exercise 

### PR DESCRIPTION
**Pull request by jaakkonarhi**
# Testing
<table>
  <tr>
    <th>unit</th>
    <th>integration</th>
    <th>e2e</th>
    <th>manual</th>
  </tr>
  <tr>
    <td>
      <ul>
        <li>- [ ] new <b>unit</b> tests created</li>
        <li>- [x] all <b>unit</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>integration</b> tests created</li>
        <li>- [x] all <b>integration</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [ ] new <b>e2e</b> tests created</li>
        <li>- [x] all <b>e2e</b> tests pass</li>
      </ul>
    </td>
    <td>
      <ul>
        <li>- [x] <b>manual</b> testing went well</li>
      </ul>
    </td>
  </tr>
</table>  
  
# Have you updated the [TESTING.md](https://github.com/Aalto-LeTech/intellij-plugin/blob/master/TESTING.md) or other relevant documentation on _your_ branch?

- [ ] Yes.
- [ ] Not yet. I will do it next.
- [x] Not relevant.

### CODE REVIEW

File: src/main/java/fi/aalto/cs/apluscourses/model/Points.java [hunk 1]
```diff
@@ -60,6 +60,10 @@ public Points(@NotNull Map<Long, List<Long>> exercises,
     return exercises.getOrDefault(exerciseGroupId, Collections.emptyList());
   }
 
+  public int getExercisesAmount() {
+    return exercises.values().stream().mapToInt(List::size).sum();
+  }
+
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/model/Points.java:66]: Should this be calculated only once (either in constructor or lazily when first needed)?

File: src/main/java/fi/aalto/cs/apluscourses/ui/utils/Bindable.java [hunk 2]
```diff
@@ -26,8 +26,8 @@ public Bindable(@NotNull T target, @NotNull BiConsumer<T, S> targetSetter) {
    */
   public Bindable(@NotNull T target, @NotNull BiConsumer<T, S> targetSetter, boolean lateInvoke) {
     this.target = target;
-    this.targetSetter = lateInvoke
-            ? (t, s) -> SwingUtilities.invokeLater(() -> targetSetter.accept(t, s)) : targetSetter;
+    this.targetSetter = lateInvoke ? (t, s) -> ApplicationManager.getApplication()
+        .invokeLater(() -> targetSetter.accept(t, s)) : targetSetter;
```
Comment (OlliKiljunen) [inline src/main/java/fi/aalto/cs/apluscourses/ui/utils/Bindable.java:30]: Just interested: did `SwingUtilities::invokeLater` not work?

**Review by OlliKiljunen — APPROVED** [review]
Good code, thank you!

File: src/main/java/fi/aalto/cs/apluscourses/ui/utils/Bindable.java (hunk 2 above)
Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/ui/utils/Bindable.java:30]: I think ApplicationManager is what intellij wants us to use

Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/ui/utils/Bindable.java:30]: the SwingUtilities one doesn't work all the time

File: src/main/java/fi/aalto/cs/apluscourses/model/Points.java (hunk 1 above)
Comment (jaakkonarhi) [reply src/main/java/fi/aalto/cs/apluscourses/model/Points.java:66]: sure

**Review by stellatsv — APPROVED** [review]
Awesome!

**Pull request merged**